File: benchmarks/tasks/pattern_match/pattern_match.ts
// SPDX-License-Identifier: MIT-0

type Shape =
    | { type: "Circle"; r: number }
    | { type: "Rectangle"; w: number; h: number }
    | { type: "Triangle"; b: number; h: number };

function area(shape: Shape): number {
    switch (shape.type) {
        case "Circle": return 3.14 * shape.r * shape.r;
        case "Rectangle": return shape.w * shape.h;
        case "Triangle": return 0.5 * shape.b * shape.h;
    }
}

function fmt(n: number): string { return n === Math.floor(n) ? String(Math.floor(n)) : String(n); }

const shapes: { name: string; shape: Shape }[] = [
    { name: "Circle", shape: { type: "Circle", r: 5.0 } },
    { name: "Rectangle", shape: { type: "Rectangle", w: 4.0, h: 6.0 } },
    { name: "Triangle", shape: { type: "Triangle", b: 6.0, h: 5.0 } },
];

for (const { name, shape } of shapes) {
    console.log(`${name}: ${fmt(area(shape))}`);
}

File: benchmarks/tasks/pattern_match/pattern_match.py
# SPDX-License-Identifier: MIT-0

def area(shape):
    if shape[0] == "Circle":
        r = shape[1]
        return 3.14 * r * r
    elif shape[0] == "Rectangle":
        w, h = shape[1], shape[2]
        return w * h
    elif shape[0] == "Triangle":
        b, h = shape[1], shape[2]
        return 0.5 * b * h

shapes = [
    ("Circle", "Circle", 5.0),
    ("Rectangle", "Rectangle", 4.0, 6.0),
    ("Triangle", "Triangle", 6.0, 5.0),
]

for name, *args in [("Circle", 5.0), ("Rectangle", 4.0, 6.0), ("Triangle", 6.0, 5.0)]:
    shape = (name, *args)
    a = area(shape)
    result = int(a) if a == int(a) else a
    print(f"{name}: {result}")
